core/raftlease/fsm.go

```diff
@@ -167,12 +167,7 @@ func (f *FSM) revoke(key lease.Key, holder string) *response {
 		return invalidResponse()
 	}
 	for gKey, entries := range f.groups {
-		for k := range entries {
-			if key != k {
-				continue
-			}
-			delete(entries, key)
-		}
+		delete(entries, key)
 		if len(entries) == 0 {
 			delete(f.groups, gKey)
 		}
```

Comment:
You shouldn't need to iterate at all - the two levels of maps are effectively: 
groups[(namespace, model-uuid)] -> entries[lease] -> (holder, start, duration)

So this should work:
```
delete(entries, key)
if len(entries) == 0 {
    delete(f.groups, groupKeyFor(key))
}